ui_framework/components/button/kui_button.js

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import classnames from 'classnames';
+
+import { KuiButtonLink } from './kui_button_link';
+import { KuiTooltip } from '../tooltip/kui_tooltip';
+
+export function KuiButton(props) {
+  if (props.href && props.onClick) {
+    throw new Error('Only href or onClick can be defined, not both');
+  }
+
+  if (props.hide) return null;
+
+  const classes = classnames('kuiButton', props.className);
+
+  if (props.href) {
+    return <KuiButtonLink {...props} />;
+  } else {
+    return <KuiTooltip text={ props.tooltip }>
+      <button
+        className={ classes }
+        aria-label={ props.tooltip }
+        onClick={ props.onClick }
+      >
+        { props.children }
+      </button>
+    </KuiTooltip>;
```

Comment:
The migration would work the same with the current implementation since the logic to not render the ReactTooltip is inside KuiTooltip.  I did it this way to avoid adding the conditional logic to everything that might have a tooltip.  I guess I don't feel very strongly either way, so I can move it out.